pyproject.toml

```diff
@@ -76,6 +76,7 @@ exclude_lines = [
 ]
 
 [tool.mypy]
+python_version=3.12
```

Comment:
If I'm reading things correctly, this setting allows for Mypy to "use" a particular version of Python no matter what is actually installed in the environment. I think this is a better setting to use than having the conditional live in the CI workflow because this (should!) ensure that all developers get a similar Mypy experience. I've been bit by something like this before